internal/index/autoindex.go

```diff
@@ -65,50 +91,58 @@ func NewAutoIndexer(s Scheduler, paths []string, opts ...Option) (*autoIndexer,
 		}
 
 		onFileChanged := func(name string) {
+			if isDone() {
+				return
+			}
 			if settings.filter(name) {
 				s.Schedule(name, 10*time.Second)
 			}
 		}
 		go w.Watch(onFileChanged)
 	}
-
-	for _, path := range paths {
-		_ = index(path, settings.MaxDepth, perFileFn, perDirFn)
-	}
+	a.perFileFn = perFileFn
+	a.perDirFn = perDirFn
 
 	return a, nil
 }
 
-func (a *autoIndexer) Close() {
-	a.watcher.Close()
-}
-
-func index(path string, maxDepth int, perFileFn func(string), perDirFn func(string) bool) error {
+func (a *autoIndexer) Index(path string) error {
 	info, err := os.Stat(path)
 	if err != nil {
-		return fmt.Errorf(`: %s: %w`, path, err)
+		return fmt.Errorf("failed to get file info: %w", err)
 	}
 	if info.IsDir() {
-		walk(path, 0, maxDepth, perFileFn, perDirFn)
+		if err := walk(path, 0, a.settings.MaxDepth, a.perFileFn, a.perDirFn); err != nil {
+			return err
+		}
 	} else {
-		perFileFn(path)
+		a.perFileFn(path)
 	}
 	return nil
 }
 
-func walk(dir string, currentDepth int, maxDepth int, perFileFn func(string), perDirFn func(string) bool) {
+func (a *autoIndexer) Close() {
+	close(a.done)
+	a.watcher.Close()
+}
+
+func walk(dir string, currentDepth int, maxDepth int, perFileFn func(string), perDirFn func(string) bool) error {
```

Comment:
could ``walk`` be an ``autoIndexer`` method? i.e: ``a.walk(path, currentDepth + 1)``